src/main/java/io/fabric8/maven/docker/service/helper/StartContainerExecutor.java

```diff
@@ -41,39 +42,41 @@
 
     private StartContainerExecutor(){}
 
-    public String startContainers() throws IOException, ExecException {
+    public AbstractMap.SimpleEntry<String, Properties> startContainers() throws IOException, ExecException {
         final Properties projProperties = projectProperties;
 
         final String containerId = hub.getRunService().createAndStartContainer(imageConfig, portMapping, gavLabel, projProperties, basedir, containerNamePattern, buildDate);
 
         showLogsIfRequested(containerId);
-        exposeContainerProps(containerId);
+        Properties exposedProperties = exposeContainerProps(containerId);
         waitAndPostExec(containerId, projProperties);
 
-        return containerId;
+        return new AbstractMap.SimpleEntry(containerId, exposedProperties);
     }
 
-    private void exposeContainerProps(String containerId)
+    private Properties exposeContainerProps(String containerId)
         throws DockerAccessException {
         String propKey = getExposedPropertyKeyPart();
+        Properties exposedProperties = new Properties();
 
         if (StringUtils.isNotEmpty(exposeContainerProps) && StringUtils.isNotEmpty(propKey)) {
             Container container = hub.getQueryService().getMandatoryContainer(containerId);
 
             String prefix = addDot(exposeContainerProps) + addDot(propKey);
-            projectProperties.put(prefix + "id", containerId);
+            exposedProperties.put(prefix + "id", containerId);
```

Comment:
what happens to `projectProperties` in this class. Is this needed anymore ? If not maybe its better to remove it then ?

Reply:
I think it's still used, it's being passed to `createAndStartContainer`:
https://github.com/fabric8io/docker-maven-plugin/blob/493aa8be14c642fc148c0a0eed717945ef9ef40f/src/main/java/io/fabric8/maven/docker/service/helper/StartContainerExecutor.java#L47

It seems to be consumed by `ContainerCreateConfig#environment()` :
https://github.com/fabric8io/docker-maven-plugin/blob/493aa8be14c642fc148c0a0eed717945ef9ef40f/src/main/java/io/fabric8/maven/docker/access/ContainerCreateConfig.java#L75-L77